craft_providers/bases/buildd.py

```diff
@@ -99,6 +100,47 @@ def __init__(
 
         self.hostname = hostname
 
+    def _ensure_os_compatible(
+        self, *, executor: Executor, deadline: Optional[float]
+    ) -> None:
+        """Ensure OS is compatible with Base.
+
+        :raises BaseCompatibilityError: if instance is incompatible.
+        :raises BaseConfigurationError: on other unexpected error.
+        """
+        try:
+            # Replace encoding errors if it somehow occurs with utf-8. This
+            # doesn't need to be perfect for checking compatibility.
+            _check_deadline(deadline)
+            proc = executor.execute_run(
+                command=["cat", "/etc/os-release"],
+                capture_output=True,
+                check=True,
+                text=True,
+                encoding="utf-8",
+                errors="replace",
+            )
+        except subprocess.CalledProcessError as error:
+            raise BaseConfigurationError(
+                brief="Failed to read /etc/os-release.",
+                details=errors.details_from_called_process_error(error),
+            ) from error
+
+        os_release = parse_os_release(proc.stdout)
+
+        os_id = os_release.get("NAME")
+        if os_id != "Ubuntu":
+            raise BaseCompatibilityError(
+                reason=f"Exepcted OS 'Ubuntu', found {os_id!r}"
+            )
```

Comment:
If os-release calls it NAME, I would keep that _name_ :-)

```suggestion
        os_name = os_release.get("NAME")
        if os_name != "Ubuntu":
            raise BaseCompatibilityError(
                reason=f"Exepcted OS 'Ubuntu', found {os_name!r}"
            )
```